core/arch/arm/kernel/generic_boot.c

```diff
@@ -927,6 +933,21 @@ static void init_external_dt(unsigned long phys_dt)
 	IMSG("Non-secure external DT found");
 }
 
+static int mark_optee_core_as_reserved(struct dt_descriptor *dt)
+{
+	vaddr_t coremem_start = CFG_TZDRAM_START;
+	vaddr_t coremem_end = CFG_TZDRAM_START + CFG_TZDRAM_SIZE;
+
+	if (coremem_start != coremem_end) {
+		return add_res_mem_dt_node(dt, "optee_core",
```

Comment:
Should we replace `"optee"` with `"optee_shm"` at line 883?

Reply:
Taking a very quick glance into the kernel driver, it relies on information from optee for the shared memory area, so a rename should not break the driver there.
But I'm not entirely sure.